### PR TITLE
Merge paradox/scala/io-udp.md and java/io-udp.md

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #23052

## Rendered HTML:

- https://richard-akka-doc.firebaseapp.com/scala/io-udp.html
- https://richard-akka-doc.firebaseapp.com/java/io-udp.html

## I split this in two commits:

- **1st commit:** Identical contents in scala/io-udp.md and java/io-udp.md -> **this is to take the diff both in scala/io-udp.md and java/io-udp.md**

  - [scala/io-udp.md  (prev) vs. scala/io-udp.md (after)](https://github.com/richard-imaoka/akka/commit/c0055e0ef41d70126e52f79adcae370a1b807d34#diff-6659bae3006f90fbb3cc3f54ab9f1cf1)
  - [java/io-udp.md (prev) vs. java/io-udp.md (after)](https://github.com/richard-imaoka/akka/commit/c0055e0ef41d70126e52f79adcae370a1b807d34#diff-4b84afe6b22f04e192f9a1f4c7a99ed6)

- **2nd commit:** Replace java/io-udp.md with a symlink to scala/io-udp.md

### CODE REVIEW

File: akka-docs/src/main/paradox/scala/io-udp.md [hunk 1]
```diff
@@ -17,12 +17,16 @@ offered using distinct IO extensions described below.
...
+:  @@snip [UdpDocSpec.scala]($code$/scala/docs/io/UdpDocSpec.scala) { #sender }
+
+Java
+:  @@snip [UdpDocTest.java]($code$/java/jdocs/io/UdpDocTest.java) { #sender }
 
 The simplest form of UDP usage is to just send datagrams without the need of
 getting a reply. To this end a “simple sender” facility is provided as
 demonstrated above. The UDP extension is queried using the
-`SimpleSender` message, which is answered by a `SimpleSenderReady`
+@scala[`SimpleSender` message]@java[`UdpMessage.simpleSender` method], which is answered by a `SimpleSenderReady`
```
Comment (2m) [inline akka-docs/src/main/paradox/scala/io-udp.md]: In Java the extension is still queried with a message (not a method), but the message is created by calling the factory that is mentioned here in the docs.

File: akka-docs/src/main/paradox/scala/io-udp.md [hunk 2]
```diff
@@ -38,23 +42,27 @@ want to close the ephemeral port the sender is bound to.
...
-@@snip [UdpDocSpec.scala]($code$/scala/docs/io/UdpDocSpec.scala) { #listener }
+Scala
+:  @@snip [UdpDocSpec.scala]($code$/scala/docs/io/UdpDocSpec.scala) { #listener }
+
+Java
+:  @@snip [UdpDocTest.java]($code$/java/jdocs/io/UdpDocTest.java) { #listener }
 
 If you want to implement a UDP server which listens on a socket for incoming
-datagrams then you need to use the `Bind` command as shown above. The
+datagrams then you need to use the @scala[`Bind` command]@java[`UdpMessage.bind` method] as shown above. The
```
Comment (2m) [inline akka-docs/src/main/paradox/scala/io-udp.md]: Same here as above. `UdpMessage.bind` is a factory for `bind` message.

File: akka-docs/src/main/paradox/scala/io-udp.md [hunk 3]
```diff
@@ -38,23 +42,27 @@ want to close the ephemeral port the sender is bound to.
...
 local address specified may have a zero port in which case the operating system
 will automatically choose a free port and assign it to the new socket. Which
 port was actually bound can be found out by inspecting the `Bound`
 message.
 
 The sender of the `Bound` message is the actor which manages the new
-socket. Sending datagrams is achieved by using the `Send` message type
-and the socket can be closed by sending a `Unbind` command, in which
+socket. Sending datagrams is achieved by using the @scala[`Send` message type]@java[`UdpMessage.send` method]
+and the socket can be closed by sending a @scala[`Unbind` command]@java[`UdpMessage.unbind` method], in which
```
Comment (2m) [inline akka-docs/src/main/paradox/scala/io-udp.md]: Same here. Message versus method.

File: akka-docs/src/main/paradox/scala/io-udp.md [hunk 4]
```diff
@@ -86,15 +98,27 @@ Akka provides a way to control various options of `DatagramChannel` through the
...
 Another socket option will be needed to join a multicast group.
 
-@@snip [ScalaUdpMulticast.scala]($code$/scala/docs/io/ScalaUdpMulticast.scala) { #multicast-group }
+Scala
+:  @@snip [ScalaUdpMulticast.scala]($code$/scala/docs/io/ScalaUdpMulticast.scala) { #multicast-group }
+
+Java
+:  @@snip [JavaUdpMulticast.java]($code$/java/jdocs/io/JavaUdpMulticast.java) { #multicast-group }
+
+Socket options must be provided to @scala[`UdpMessage.Bind` message]@java[`UdpMessage.bind` method].
```
Comment (2m) [inline akka-docs/src/main/paradox/scala/io-udp.md]: Message vs method.

**Review by 2m — COMMENTED** [review]
Looking good. Just need to s/method/message on the java side, as it is still messages that are being sent on the Java side. However in the documentation we refer to the factories, that create these messages.

**Comment by richard-imaoka** [comment]
Sure, replaced "method" with "message" as advised.

**Review by 2m — APPROVED** [review]

File: akka-docs/src/main/paradox/java/io-udp.md [hunk 5]
```diff
@@ -1,100 +0,0 @@
-# Using UDP
```
Comment (ktoso) [inline akka-docs/src/main/paradox/java/io-udp.md]: that's how github shows a symlink?

**Review by ktoso — APPROVED** [review]
LGTM, thanks!

File: akka-docs/src/main/paradox/java/io-udp.md (hunk 5 above)
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/java/io-udp.md]: Yeah, see https://github.com/akka/akka/pull/23096/files as another example.

**Pull request merged**